irispy/iris_tools.py

```diff
@@ -212,6 +220,97 @@ def get_iris_response(pre_launch=False, response_file=None, response_version=Non
     return iris_response
 
 
+def fit_iris_xput(time_obs, time_cal_coeffs, cal_coeffs):
+    """
+    To calculate the coefficients of best-fit time function for throughput,
+    for which there are two modes:
+    1. Perform fit: supply xput and single element ``cal_coeffs``.
+    2. Apply fit: supply full ``cal_coeffs``.
+
+    Parameters
+    ----------
+    time_obs: a numpy array of floats
+    - Times of datapoints.
+
+    time_cal_coeffs: a numpy array of floats
+    - Start and end times of intervals of constant ``cal_coeffs[i]``.
+
+    cal_coeffs: a numpy array of floats
+    - Coefficients of best-fit function.
+
+    Returns
+    -------
+    Yields the values of fit for times ``time_obs``.
+
+    """
+    # Set the reference time.
+    t_ref = parse_time('1979-01-01', time=1)
+    
+    if isinstance(time_cal_coeffs, np.ndarray):
+        if isinstance(cal_coeffs, np.ndarray):
+            size_time_cal_coeffs = time_cal_coeffs.shape
+            size_cal_coeffs = cal_coeffs.shape
+    
+    # Some time transformations.
+    t_cal_coeffs = []
+    t_cal_coeffs_flat = list(time_cal_coeffs.flatten())
+    for element in t_cal_coeffs_flat:
+        tmp = parse_time(float(element), utime=1) - t_ref
+        t_element = tmp.total_seconds()
+        t_cal_coeffs.append(t_element)
+    t_cal_coeffs = np.array(t_cal_coeffs).reshape(5, 2)
+
+    for i in time_obs:
+        aux_cal_coeffs = np.zeros(2*size_time_cal_coeffs[0])
+        t_obs = parse_time(i, utime=1) - t_ref
+        t_obs = t_obs.total_seconds()
+
+    if size_time_cal_coeffs[1] != 2 or size_cal_coeffs[1] < 2:
+        # Raise ValueError as time coefficient have the wrong format.
+        raise ValueError("Incorrect number of elements either in time_cal_coeffs or in cal_coeffs.")
+    # Exponent for transition between exp.decay intervals.
+    transition_exp = 1.5
+
+    fit_out = np.zeros(len(list(time_obs)), dtype=np.float64)
```

Comment:
```time_obs``` shouldn't have to be converted to get to ```len```.

Reply:
No, wouldn't work since:
```
In [3]: fit_iris_xput(time_obs, time_cal_coeffs0, cal_coeffs0)                  
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-574283a177a7> in <module>
----> 1 fit_iris_xput(time_obs, time_cal_coeffs0, cal_coeffs0)

~/irispy-git/irispy/iris_tools.py in fit_iris_xput(time_obs, time_cal_coeffs, cal_coeffs)
    272     transition_exp = 1.5
    273 
--> 274     fit_out = np.zeros(list(time_obs), dtype=np.float64)
    275 
    276     # Looking for the closest time in the calibration time intervals.

TypeError: 'datetime.datetime' object cannot be interpreted as an integer
```